src/btree.c

```diff
@@ -3248,29 +3252,33 @@ btree_print_subtree(cache                 *cc,
    if (node.hdr->height > 0) {
       for (idx = 0; idx < node.hdr->num_entries; idx++) {
          btree_print_subtree(
-            cc, cfg, btree_get_child_addr(cfg, node.hdr, idx), stream);
+            cc, cfg, btree_get_child_addr(cfg, node.hdr, idx), log_handle);
       }
    }
    btree_node_unget(cc, cfg, &node);
 }
 
 void
-btree_print_tree(cache *cc, btree_config *cfg, uint64 root_addr)
+btree_print_tree(cache               *cc,
+                 btree_config        *cfg,
+                 uint64               root_addr,
+                 platform_log_handle *log_handle)
 {
-   platform_open_log_stream();
-   btree_print_subtree(cc, cfg, root_addr, stream);
-   platform_close_log_stream(PLATFORM_DEFAULT_LOG_HANDLE);
+   btree_print_subtree(cc, cfg, root_addr, log_handle);
 }
 
 void
-btree_print_tree_stats(cache *cc, btree_config *cfg, uint64 addr)
+btree_print_tree_stats(cache               *cc,
+                       btree_config        *cfg,
+                       uint64               addr,
+                       platform_log_handle *log_handle)
 {
    btree_node node;
    node.addr = addr;
    btree_node_get(cc, cfg, &node, PAGE_TYPE_BRANCH);
 
-   platform_log("Tree stats: height %u\n", node.hdr->height);
-   cache_print_stats(cc);
+   platform_default_log("Tree stats: height %u\n", node.hdr->height);
+   cache_print_stats(cc, log_handle);
```

Comment:
btw, I pondered on my own request to consider changing `log_handle` to be the 1st arg. That would cascade to cache-printing function, also, whose interface before this change was:

```
387 cache_print_stats(cache *cc)
```

And after this change becomes:

```
388 cache_print_stats(cache *cc, platform_log_handle *log_handle)
```

I reasoned that as these print methods take anywhere from 1, 2, 3 parameters, it will be [more] consistent to say that the 1st arg is **always**-always the output log-handle / stream.

I am saying: While we are roto-tilling these interfaces, again -- it seems to me to be the right time to move to this 1st-arg-convention I am suggesting here.